modules/gateways/veritrans.php

```diff
@@ -360,10 +360,24 @@ function veritrans_link($params)
         $snapToken = $snap_transaction->token;
         $redirect_url = $snap_transaction->redirect_url;
         // error_log(" ############# TOKEN ::: ".$snapToken);
+
+        // save snap redirect url to invoice notes
+        $command = 'UpdateInvoice';
+        $postData = array(
+            'invoiceid' => $invoiceId,
+            'notes' => $redirect_url
+        );
+        $results = localAPI($command, $postData);
```

Comment:
This is not the best approach, you are replacing `notes` attributes without checking if there are any previous value assigned to it. Think about this scenario:
- What happen if customer/merchant already put their own order notes on the invoice? It will be gone and unexpectedly replaced with Snap URL.

How should you handle this? My suggestion is don't just **replace**, but **append** Snap URL behind existing value. Also put a consistent separator string. For example:
```
'notes'` => "$existing_notes #PayURL: $snap_redirect_url"
```
Also:
- Be ware of max string limit, what is max string limit for the notes? will the API call fail if it reach max limit? how should you handle the error?
- Also make sure to put a try-catch around this UpdateInvoice, we don't want to break the overall payment flow simply because fail to update Invoice.

Please update accordingly, thanks.